azure-iot-device/tests/common/test_mqtt_transport.py

```diff
@@ -947,6 +949,24 @@ def test_does_not_set_thread_to_none_on_exception_in_non_paho_thread(
 
 @pytest.mark.describe("MQTTTransport - OCCURANCE: Disconnect Completed")
 class TestEventDisconnectCompleted(object):
+    @pytest.fixture
+    def collected_transport_weakref(self, mock_mqtt_client):
+        # return a weak reference to an MQTTTransport that has already been collected
+        transport = MQTTTransport(
+            client_id=fake_device_id, hostname=fake_hostname, username=fake_username
+        )
+        transport_weakref = weakref.ref(transport)
+        transport = None
+        gc.collect(2)
```

Comment:
>gc.collect(2) [](start = 8, length = 13)

can u add a comment explaining the 2 ? this means that we are doing a full collection of the oldest objects that are residing in generation 2 ?